packages/theme/src/cli/utilities/theme-environment/html.ts

```diff
@@ -31,6 +32,17 @@ export function getHtmlHandler(theme: Theme, ctx: DevServerContext) {
 
         assertThemeId(response, html, String(theme.id))
 
+        if (ctx.options.errorOverlay !== 'silent' && ctx.localThemeFileSystem.uploadErrors.size > 0) {
+          html = getErrorPage({
```

Comment:
I was going to say that maybe we should set the status here to 500. However, what we are showing is the summary of the upload errors, not the rendering errors... so perhaps we should not change the status? :confused_parrot:
Ideally we'd just inject a modal overlay on top of the rendered page so that it makes sense to use status 200 and still show errors. But that doesn't need to be done now of course.
Thoughts? cc @karreiro 

-- Thinking more about this, the observed result from the user is an error page so I think we should probably return status 500 regardless of the nature of the error until we move this to a modal 🤔

Reply:
Do you think we should try to expand the accepted types of the catch block below and use that?

Reply:
Actually, what if you do this check before calling `render` and just return the error without reaching SFR? Since it's not a real overlay yet, we don't need the actual storefront document, right?